hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTxTest.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    protected final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.shutdownAll();
+    }
+
+    @Test
+    public void test_whenEmptyMap_thenPopulatedNearCacheShouldReturnNull_neverNULL_OBJECT() {
+        int size = 10;
+        String mapName = randomMapName();
+
+        NearCacheConfig nearCacheConfig = newNearCacheConfig();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
+
+        HazelcastInstance client = getNearCachedClient(mapName, config, nearCacheConfig);
+
+        // populate map
+        TransactionContext transactionContext = client.newTransactionContext();
+        transactionContext.beginTransaction();
+        TransactionalMap<Integer, Integer> map = transactionContext.getMap(mapName);
+        for (int i = 0; i < size; i++) {
+            // populate Near Cache
+            assertNull(map.get(i));
+            // fetch value from Near Cache
+            assertNull(map.get(i));
+        }
+        transactionContext.commitTransaction();
+    }
+
+    @Test
+    public void testBasicUsage() {
+        int mapSize = 5000;
+        String mapName = "testBasicUsage";
+
+        NearCacheConfig nearCacheConfig = newNearCacheConfig().setInvalidateOnChange(true);
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
+
+        HazelcastInstance client = getNearCachedClient(mapName, config, nearCacheConfig);
+
+        TransactionContext transactionContext = client.newTransactionContext();
+        transactionContext.beginTransaction();
+        TransactionalMap<Integer, Integer> map = transactionContext.getMap(mapName);
+        populateMap(map, mapSize);
+        transactionContext.commitTransaction();
+
+        IMap<Integer, Integer> instanceMap = client.getMap(mapName);
+        for (int i = 0; i < mapSize; i++) {
+            assertNotNull(instanceMap.get(i));
+        }
+
+        transactionContext = client.newTransactionContext();
+        transactionContext.beginTransaction();
+        map = transactionContext.getMap(mapName);
+        for (int i = 0; i < mapSize; i++) {
+            map.put(i, i * 2);
+        }
+        transactionContext.commitTransaction();
+
+        IMap<Object, Object> m = client.getMap(mapName);
+        for (int i = 0; i < mapSize; i++) {
+            assertNotNull(m.get(i));
+        }
+
+        assertNearCacheNotEmpty(mapName, client);
+
+        transactionContext = client.newTransactionContext();
+        transactionContext.beginTransaction();
+        map = transactionContext.getMap(mapName);
+        for (Integer key : map.keySet()) {
+            map.remove(key);
+        }
+        transactionContext.commitTransaction();
+        assertNearCacheEmptyEventually(mapName, client);
+    }
+
+    @Override
+    protected Config getConfig() {
```

Comment:
I would not call this method 'getConfig' since you are not 'getting' something, you are creating a new config. So newConfig or createConfig would be better describing the purpose of this method.